libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -249,16 +258,33 @@ - (SFRestRequest *)requestForCreateWithObjectType:(NSString *)objectType
 - (SFRestRequest *)requestForUpdateWithObjectType:(NSString *)objectType
                                          objectId:(NSString *)objectId
                                            fields:(NSDictionary *)fields {
+    return [self requestForUpdateWithObjectType:objectType objectId:objectId fields:fields ifUnmodifiedSinceDate:nil];
+}
+
+- (SFRestRequest *)requestForUpdateWithObjectType:(NSString *)objectType
+                                         objectId:(NSString *)objectId
+                                           fields:(NSDictionary *)fields
+                            ifUnmodifiedSinceDate:(NSDate *) ifUnmodifiedSinceDate {
+
     NSString *path = [NSString stringWithFormat:@"/%@/sobjects/%@/%@", self.apiVersion, objectType, objectId];
-    return [SFRestRequest requestWithMethod:SFRestMethodPATCH path:path queryParams:fields];
+    SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodPATCH path:path queryParams:fields];
+    [request setHeaderValue:[SFRestAPI getHttpStringFomFromDate:ifUnmodifiedSinceDate] forHeaderName:kSFRestIfUnmodifiedSince];
+    return request;
 }
 
 - (SFRestRequest *)requestForUpsertWithObjectType:(NSString *)objectType
                                   externalIdField:(NSString *)externalIdField
                                        externalId:(NSString *)externalId
                                            fields:(NSDictionary *)fields {
-    NSString *path = [NSString stringWithFormat:@"/%@/sobjects/%@/%@/%@", self.apiVersion, objectType, externalIdField, externalId];
-    return [SFRestRequest requestWithMethod:SFRestMethodPATCH path:path queryParams:fields];
+    NSString *path = [NSString stringWithFormat:@"/%@/sobjects/%@/%@/%@",
```

Comment:
Since v37, upsert with external id null is allowed - it does a create (but post must be used).